### ISSUE TITLE
Documentation about Registration::setEuclideanFitnessEpsilon() incorrect

### ISSUE THREAD

**Issue by themightyoarfish** [PointCloudLibrary/pcl#5868]
https://github.com/PointCloudLibrary/pcl/blob/81b92a50f21b63c8a3bce9c900dfe6d247ee51e4/registration/include/pcl/registration/registration.h#L395

The doc claims that the epsilon refers to the "maximum allowed Euclidean error between two consecutive steps" in the registration. This does not seem to be correct at least for `IterativeClosestPoint`, where the value is passed to `DefaultConvergenceCriteria` where it is used as the relative reduction in MSE between two steps. i.e. a value of 0.01 does not mean "converge when current MSE < 0.01", but rather "converge when reduction in MSE from last iteration is less than 1% of MSE from last iteration".

I'm not sure if the documentation should be updated, since other subclasses may use it differently, or if the entire function should move to the icp class, since only ICP and JointICP seem to actually use it.

**Comment by mvieth** [comment]
I agree that the description is not so great. Feel free to suggest a rewording, based on how ICP and JointICP use it.
